### PR TITLE
Adds online functionality to the book-build cmd & requirement's check.

### PR DESCRIPTION
**Pull request by Cortesj**
Adds additional functionality to docker_tools. Two additional parameters added "--online" & "--username" which means that specified book will download from online repository if "--online" flag is specified. The "--username" pulls from specified GitHub Username. The appropriate checks have been added to check for existing repositories/books and verify that remote repository is a valid repository. If username specified doesn't have valid repository then default to _RunestoneInteractive_ as default username to attempt cloning the book.

Additional functionally added is checking if book has any requirements prior to build & deployment. It will attempt to find a requirements.txt file within sub_book_name. If nothing is found then it will skip step and proceed with build & deployment.

**TL:DR**
- Adds requirements.txt check to book-build
- Adds flag "--online" to clone book from Remote Repository
- Adds flag "--username" to clone book from specific GitHub user when "--online" is specified
- Check if remote repository is available and following certain actions in case of error reaching remote repo.

Pertains to issue #1775

### CODE REVIEW

**Comment by bjones1** [comment]
Jose,

I'd suggest adding this functionality to `rsmanage` instead of the Docker tools -- that way, users can install books on a stand-along version (BookServer only) or on the combined web2py + BookServer setup.

**Comment by bnmnetp** [comment]
@bjones1 - I don't understand your comment, building a book is part of the docker_tools.py script.  It is not part of rsmanage.

I know you have proposed, elsewhere, that we migrate the build command to rsmanage but that hasn't happened.

**Comment by Cortesj** [comment]
@bjones1 Could you explain more on rsmanage? I haven't seen that script yet. I am assuming its currently within the master build. I will be taking a look at it to see how I can implement it there.

**Comment by bjones1** [comment]
@Cortesj, sorry, I missed this earlier. Take a look at #1810 -- does that help? Also, have you joined https://runestoneacademy.discourse.group/? There's a fair amount of discussion there.

**Comment by Cortesj** [comment]
I noticed that a discourse group was created and joined it fairly recently. I guess I have more reading to catch up on! I might close the PR considering that movement to different functionality to different portions of the system.

**Comment by bjones1** [comment]
I agree -- I really like this code, but it belongs in `rsmanage`.
